sdk/core/azure-core/tests/test_authentication.py

```diff
@@ -146,3 +146,40 @@ def test_key_vault_regression():
     policy._token = AccessToken(token, time.time() + 3600)
     assert not policy._need_new_token
     assert policy._token.token == token
+
+def test_azure_key_credential_policy():
+    """Tests to see if we can create an AzureKeyCredentialPolicy"""
+    def verify_authorization_header(request):
+        assert request.http_request.headers["api_key"] == "test_key"
+    api_key = "test_key"
+    key_header = "api_key"
+
+    transport=Mock()
+    credential = AzureKeyCredential(api_key)
+    credential_policy = AzureKeyCredentialPolicy(credential=credential, name=key_header)
+    policies = [credential_policy, Mock(send=verify_authorization_header)]
+    pipeline = Pipeline(transport=Mock(), policies=policies)
+
+    pipeline.run(HttpRequest("GET", "https://test_key_credential"))
```

Comment:
That's not enough, you need to check that the HttpRequest headers has the key, something likle (pseudo-code):
```
req = pipeline.run(HttpRequest("GET", "https://test_key_credential"))
pipeline.run(req)
assert req.headers["api_key"] == "test_key"
```